### PR TITLE
Add feature to add Credential Type

### PR DESCRIPTION
**Pull request by nixocio**
Add feature to add Credential Type.

See: https://github.com/ansible/awx/issues/7325

add_credential_type

### CODE REVIEW

**Review by dsesami — APPROVED** [review]
Looks great!

File: awx/ui_next/src/util/yaml.test.js [hunk 1]
```diff
@@ -68,3 +68,44 @@ two: two
...
+  test('should convert yaml string to json object', () => {
+    expect(parseVariableFields(injector)).toEqual(
+      JSON.parse(yamlToJson(injector))
+    );
+  });
+
+  test('should convert json string to json object', () => {
+    expect(parseVariableFields(input)).toEqual(JSON.parse(input));
+  });
+});
```
Comment (jlmitch5) [inline awx/ui_next/src/util/yaml.test.js:111]: nice tests

File: awx/ui_next/src/util/yaml.js [hunk 2]
```diff
@@ -35,3 +35,16 @@ export function isJson(jsonString) {
 
   return typeof value === 'object' && value !== null;
 }
+
+export function parseVariableFields(variableField) {
```
Comment (jlmitch5) [inline awx/ui_next/src/util/yaml.js]: can you rename to parseVariableField

**Review by jlmitch5 — CHANGES_REQUESTED** [review]
looks good!  just a quick rename of the yaml util fn

File: awx/ui_next/src/util/yaml.js (hunk 2 above)
Comment (nixocio) [reply awx/ui_next/src/util/yaml.js]: @jlmitch5 updated per comment.

**Review by jlmitch5 — APPROVED** [review]

**Pull request merged**